eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/clustering_writer/RemappedVariantsClusteringWriterTest.java

```diff
@@ -339,137 +339,173 @@ public void test_variant_not_clustered_rs_id_exist() throws Exception {
 
-        List<SubmittedVariantEntity> expectedSubmittedVariantEntityRetainRS = submittedVariantEntities.stream()
-                .filter(sve->sve.getClusteredVariantAccession().equals(3000000006L))
-                .filter(sve->sve.getReferenceSequenceAccession().equals("asm2"))
-                .collect(Collectors.toList());
-        assertEquals(1, expectedSubmittedVariantEntityRetainRS.size());
-        assertEquals(1500, expectedSubmittedVariantEntityRetainRS.get(0).getStart());
 
         //assert clusteredVariantEntity
         List<ClusteredVariantEntity> clusteredVariantEntities = mongoTemplate.findAll(ClusteredVariantEntity.class);
-        assertEquals(3, clusteredVariantEntities.size());
-
-        List<ClusteredVariantEntity> expectedClusteredVariantEntityWithNewRS = clusteredVariantEntities.stream()
-                                        .filter(cls -> cls.getAccession().equals(3000000000L))
-                                        .collect(Collectors.toList());
-        assertEquals(1, expectedClusteredVariantEntityWithNewRS.size());
-        assertEquals(Long.valueOf(3000000000L), expectedClusteredVariantEntityWithNewRS.get(0).getAccession());
-        assertEquals(1000, expectedClusteredVariantEntityWithNewRS.get(0).getStart());
-
-        List<ClusteredVariantEntity> expectedClusteredVariantEntityWithRetainRS = clusteredVariantEntities.stream()
-                                        .filter(cls -> cls.getAccession().equals(3000000006L))
-                                        .filter(cls->cls.getAssemblyAccession().equals("asm2"))
-                                        .collect(Collectors.toList());
-        assertEquals(1, expectedClusteredVariantEntityWithRetainRS.size());
-        assertEquals(Long.valueOf(3000000006L), expectedClusteredVariantEntityWithRetainRS.get(0).getAccession());
-        assertEquals(1500, expectedClusteredVariantEntityWithRetainRS.get(0).getStart());
-
-        //assert submittedVariantOperationEntity
-        List<SubmittedVariantOperationEntity> submittedVariantOperationEntities = mongoTemplate.findAll(
-                SubmittedVariantOperationEntity.class);
-        assertEquals(1, submittedVariantOperationEntities.size());
+        assertEquals(4, clusteredVariantEntities.size());
+        assertEquals(1,clusteredVariantEntities.stream().filter(cve->cve.getAccession().equals(3000000003L)).count());
+        assertEquals(3,clusteredVariantEntities.stream().filter(cve->cve.getAccession().equals(3000000001L)).count());
+
+        //assert merge event
+        SubmittedVariantOperationEntity submittedVariantOperationEntityMerge =
+                mongoTemplate.findAll(SubmittedVariantOperationEntity.class).stream()
+                .filter(svoe->svoe.getEventType().equals(EventType.RS_MERGE_CANDIDATES))
+                .collect(Collectors.toList()).get(0);
+        assertEquals(2, submittedVariantOperationEntityMerge.getInactiveObjects().size());
```

Comment:
Would also suggest inspecting the actual objects inside the inactiveObjects array.

Reply:
added